passportValue/helpers.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Find a current  passport variable (fn) or (val) and replace it in each node in a flow (live or published
+ */
+
+function replaceAllOccurrences(fullPassportValue, currentPassportVariable, newPassportVariable) {
+  const regex = new RegExp('\\b' + currentPassportVariable + '\\b', 'g');
+  return fullPassportValue.replace(regex, newPassportVariable);
```

Comment:
Sorry - still not personally convinced this regex that protects against partial matches is better approach than plain string `replaceAll()` and I think is going to be ultimately limiting in use of this against "real" content examples!

In your example `alter` &rarr; `change` (but _not_ `alteration` &rarr; `changeation`) that this regex protects against, I think you're overlooking the option to simply pass `alter.` as the find variable then `replaceAll()` as expected !! 

In recent file naming, there were other examples of partial matches like `councilTaxBill` (find = `councilTax`) &rarr; `councilBill` (replace = `council`) that wouldn't be supported by this regex.

Reply:
I'm maybe being too risk averse with this approach. As far as I understand the regex would approach would still allow all renaming but might require more runs as matches would need to be made explicitly `councilTaxBill → councilBill` vs the freedom / risk of partial match renames. 

Passing in variables with `.` as in `alter.` is definitely something I hadn't considered and does overcome the issue I'd outlined although I guess it does require protecting against edge cases at the time of running the code and a deeper familiarity with the project types versus having that edge case protection in the code? 

Although, I'm happy to defer to your judgement though as you're much more familiar with this process and the data.

Reply:
Updated here: f073806941afb401f0575a2a4c8eb8afa649d7df